portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -18,24 +25,68 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     boolean isLoggedIn = false;
 
+    PrintWriter out = response.getWriter();
     UserService userService = UserServiceFactory.getUserService();
+    String loginUrl = userService.createLoginURL("/");
+
     if (userService.isUserLoggedIn()) {
       isLoggedIn = true;
       String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
+      String nickname = getUserNickname(userService.getCurrentUser().getUserId());
+      out.println("<p>Set your nickname here:</p>");
+      out.println("<form method=\"POST\" action=\"/nickname\">");
+      out.println("<input name=\"nickname\" value=\"" + nickname + "\" />");
+      out.println("<br/>");
+      out.println("<button>Submit</button>");
+      out.println("</form>");
+
+    
       response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + loginUrl + "\">here</a>.</p>");
     } else {
-      String urlToRedirectToAfterUserLogsIn = "/";
-      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
 
-      response.getWriter().println(isLoggedIn);
       response.getWriter().println("<p>Hello stranger.</p>");
       response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
     }
 
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/login");
+      return;
+    }
+
+    String nickname = request.getParameter("nickname");
```

Comment:
For string constants like "nickname", it would be good to make it a constant for better reuse, e.g.:

private final String NICKNAME = "nickname";

and then use NICKNAME here.